lib/stackops/apiGateway.js

```diff
@@ -188,6 +188,18 @@ module.exports = function(currentTemplate, aliasStackTemplates, currentAliasStac
 			aliasResources.push({ ApiGatewayStage: stageResource });
 		}
 
+		const baseMapping = _.assign({}, _.pickBy(stageStack.Resources, ['Type', 'AWS::ApiGateway::BasePathMapping']));
+		if (!_.isEmpty(baseMapping)) {
+			const baseMappingName = _.keys(baseMapping)[0];
+			const obj = baseMapping[baseMappingName];
+
+			delete obj.Properties.StageName;
```

Comment:
`StageName` should be set to `{ Ref: ApiGatewayStage }` here, so that it is automatically bound to the currently deploying alias stage, for which the resource is available in the alias stack of that alias.

And the whole baseMapping handling should be moved into the `if (!_.isEmpty(deployment)) {`, just after the stage resource creation at the very end of the if, because the stage resource is only created if we actually have an API deployment in the current alias.
This is important, because you even could deploy multiple aliases, one completely without an API and one with an API in the same project stage.

Reply:
done